src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java

```diff
@@ -1,107 +1,178 @@
+        LocalDateTime toTime = toDatePicker.getValue().atTime(toTimePicker.getValue().getHour(), toTimePicker.getValue().getMinute());
+
+        Date dateStart = Date.from(fromTime.atZone(ZoneId.systemDefault()).toInstant());
+        Date dateEnd = Date.from(toTime.atZone(ZoneId.systemDefault()).toInstant());
+
+        if (dateStart.after(dateEnd)) {
+            Alert errorAlert = new Alert(Alert.AlertType.ERROR);
+            errorAlert.setTitle("Error!");
+            errorAlert.setHeaderText(null);
+            errorAlert.setContentText("Error occurred while checking date.");
+            errorAlert.showAndWait();
+        } else {
+            ConfirmExportController confirmExportController = new ConfirmExportController(primaryStage);
+            try {
+                List<String> applicationsToExport = confirmExportController.showConfirmationAndGetAppList(trackingService.getApplicationsNames());
+                if (!confirmExportController.getCancelValue()) {
+                    if (wholeRangeCheckBox.isSelected()) {
+                        mainExporter.export(trackingService, applicationsToExport);
+                    } else {
+                        mainExporter.export(trackingService, applicationsToExport, fromTime, toTime);
+                    }
+                    Alert successAlert = new Alert(Alert.AlertType.INFORMATION);
+                    successAlert.setTitle("Success!");
+                    successAlert.setHeaderText(null);
+                    successAlert.setContentText("Data exported successfully! ");
+                    successAlert.showAndWait();
+                    primaryStage.close();
+                }
+            } catch (IOException e) {
+                Alert errorAlert = new Alert(Alert.AlertType.ERROR);
+                errorAlert.setTitle("Error!");
+                errorAlert.setHeaderText(null);
+                errorAlert.setContentText("Error occurred while exporting data.");
+                errorAlert.showAndWait();
+            }
+        }
+    }
+
+
+    @FXML
+    public void onToNow() {
+        if (toNowCheckBox.isSelected()) {
+            toDatePicker.setValue(now().plusYears(1));
+            toTimePicker.setValue(LocalTime.now());
+            toDatePicker.setEditable(false);
+            toTimePicker.setEditable(false);
+
+            wholeRangeCheckBox.setDisable(true);
+        } else {
+            toDatePicker.setEditable(true);
+            toTimePicker.setEditable(true);
+
+            if (!fromBeginCheckBox.isSelected()) wholeRangeCheckBox.setDisable(false);
+        }
+    }
+
+    @FXML
+    public void onFromBegin() {
+        if (fromBeginCheckBox.isSelected()) {
+            fromDatePicker.setValue(LocalDate.of(1970, 1, 1));
+            fromTimePicker.setValue(LocalTime.of(0, 0));
+            fromDatePicker.setEditable(false);
```

Comment:
maybe there is a way to get this lines together and make a good method for that? It is used very often in this file, so if there is a possibility to get some lines together and extract it to method pls do that